### PR TITLE
[RAPTOR-3430] mlops: adding mlops monitoring support

### PR DESCRIPTION
**Pull request by liororama**
# This repository is public. Do not put here any private DataRobot or customer's data: code, datasets, model artifacts, .etc.

## Summary
Adding support for MLOps monitoring in drum. This way we can monitor models running outside of DR using DR App.

Things to go in next  PR -

1. Documentation and example in the README - I consider this PR as a beta so I am not  updating the docs right now
2. Add support for java predictor and r predictor like done in the python predictors
3. Add tests to include the java and r predictors.

### CODE REVIEW

File: custom_model_runner/datarobot_drum/drum/language_predictors/base_language_predictor.py [hunk 1]
```diff
@@ -8,18 +9,72 @@
...
+        self._params = None
+        self._mlops = None
 
     def configure(self, params):
         self._custom_model_path = params["__custom_model_path__"]
         self._positive_class_label = params.get("positiveClassLabel")
         self._negative_class_label = params.get("negativeClassLabel")
+        self._params = params
+
+        pprint.pprint(params)
```
Comment (yakov-g) [inline custom_model_runner/datarobot_drum/drum/language_predictors/base_language_predictor.py:37]: This looks like debug print, clean up please.

File: custom_model_runner/datarobot_drum/drum/language_predictors/base_language_predictor.py [hunk 2]
```diff
@@ -8,18 +9,72 @@
...
+            # TODO: or add report_predictions_data that can handle a df directly..
+            # TODO: need to handle associds correctly
+
+            # mlops.report_predictions_data expect the prediction data in the following format:
+            # Regression: [10, 12, 13]
+            # Classification: [[0.5, 0.5], [0.7, 03]]
+            # In case of classification, class names are also required
+            class_names = None
+            if len(predictions.columns) == 1:
+                print(predictions.columns[0])
```
Comment (yakov-g) [inline custom_model_runner/datarobot_drum/drum/language_predictors/base_language_predictor.py:67]: Clean up

File: custom_model_runner/datarobot_drum/drum/language_predictors/python_predictor/python_predictor.py [hunk 3]
```diff
@@ -29,10 +32,19 @@ def configure(self, params):
             raise Exception("Failed to load model")
 
     def predict(self, input_filename):
+
         kwargs = {}
         if self._positive_class_label and self._negative_class_label:
             kwargs[POSITIVE_CLASS_LABEL_ARG_KEYWORD] = self._positive_class_label
             kwargs[NEGATIVE_CLASS_LABEL_ARG_KEYWORD] = self._negative_class_label
 
+        pprint.pprint(self._params)
```
Comment (yakov-g) [inline custom_model_runner/datarobot_drum/drum/language_predictors/python_predictor/python_predictor.py:41]: Clean up

**Review by yakov-g — CHANGES_REQUESTED** [review]

File: custom_model_runner/datarobot_drum/drum/args_parser.py [hunk 4]
```diff
@@ -488,8 +539,36 @@ def get_arg_parser():
...
+        CMRunnerArgsRegistry._reg_arg_monitor(batch_parser, server_parser, fit_parser)
+        CMRunnerArgsRegistry._reg_arg_deployment_id(batch_parser, server_parser, fit_parser)
+        CMRunnerArgsRegistry._reg_arg_model_id(batch_parser, server_parser, fit_parser)
+        CMRunnerArgsRegistry._reg_arg_monitor_settings(batch_parser, server_parser, fit_parser)
+
         return parser
 
+    @staticmethod
+    def verify_monitoring_options(options, parser_name):
+        if options.monitor:
```
Comment (yakov-g) [inline custom_model_runner/datarobot_drum/drum/args_parser.py:551]: What if I provide model_id, deployment_id, but don't provide monitor

**Review by yakov-g — CHANGES_REQUESTED** [review]

**Review by meltzg — APPROVED** [review]

**Review by yakov-g — APPROVED** [review]

**Comment by datarobot-jarvis** [comment]
:dollar: Total cost of executing tests in Jenkins: $**2.19** USD.

[What is this all about?](http://bit.ly/2BahK4V)

**Pull request merged**